vm/src/stdlib/os.rs

```diff
@@ -1191,6 +1191,55 @@ fn os_chmod(
     Ok(())
 }
 
+#[cfg(unix)]
+fn os_execv(
+    path: PyStringRef,
+    argv_list: PyIterable<PyStringRef>,
+    vm: &VirtualMachine,
+) -> PyResult<()> {
+    // TODO: argv_list to only accept tuples and lists
+    // message: "TypeError: execv() arg 2 must be a tuple or list"
+
+    if path.to_string().contains("\0") {
+        return Err(vm.new_value_error("embedded null byte".to_owned()));
+    }
+
+    let path = ffi::CString::new(path.as_str()).unwrap();
+
+    if argv_list.iter(vm)?.count() == 0 {
+        return Err(vm.new_value_error("execv() arg 2 must not be empty".to_owned()));
+    }
+
+    for arg in argv_list.iter(vm)? {
+        if arg?.as_str().contains("\0") {
+            return Err(vm.new_value_error("embedded null byte".to_owned()));
+        }
+    }
+
+    if argv_list
```

Comment:
This condition will be easier with vec from `Either`. Otherwise, checking len of `argv_list: Vec<ffi::CString>` after the creation would be also easy.

Reply:
used Vec<ffi::CString>.